### PR TITLE
concat widget lists

### PR DESCRIPTION
**Pull request by markhayden**
when leveraging a config with multiple widgets targeting multiple segments we were just overwriting with each loop and thus only ever showing the last widget in the array (if it matched). this pr ensures that we concatenate the array so that all widgets with proper segment matches have the ability to show.

### CODE REVIEW

**Comment by DingoEatingFuzz** [comment]
Tests?

**Comment by markhayden** [comment]
si, tests added

File: test/pathforaSpec.js [hunk 1]
```diff
@@ -476,6 +476,56 @@ describe('Pathfora', function () {
...
+
+    expect(widgetA).toBeDefined();
+    expect(widgetB).toBeDefined();
+
+    setTimeout(function () {
+      expect(widgetA.hasClass('opened')).toBeTruthy();
+      expect(widgetB.hasClass('opened')).toBeTruthy();
+      pathfora.clearAll();
+      done();
+    }, 200);
```
Comment (markhayden) [inline test/pathforaSpec.js]: there wasn't originally but i believe there might be now. its done like this in lots of places tho so going to make a ticket and fix them all at once.

Comment (DingoEatingFuzz) [inline test/pathforaSpec.js]: 👍

**Comment by DingoEatingFuzz** [comment]
:shipit:

**Pull request merged**